tests/unit/test_spin.cpp

```diff
@@ -676,10 +676,11 @@ SECTION("Closed-shell spintrace CCD") {
       auto pno_ccd_energy_sf =
           closed_shell_CC_spintrace(pno_ccd_energy_so_as_sum);
       REQUIRE_THAT(pno_ccd_energy_sf,
-                   SimplifiesTo("2 g{a1<i1,i2>,a2<i1,i2>;i1,i2}:N-C "
-                                "t{i1,i2;a1<i1,i2>,a2<i1,i2>}:N-C - "
+                   SimplifiesTo("- "
                                 "g{a1<i1,i2>,a2<i1,i2>;i1,i2}:N-C "
-                                "t{i1,i2;a2<i1,i2>,a1<i1,i2>}:N-C"));
+                                "t{i1,i2;a2<i1,i2>,a1<i1,i2>}:N-C"
+                                " + 2 g{a1<i1,i2>,a2<i1,i2>;i1,i2}:N-C "
+                                "t{i1,i2;a1<i1,i2>,a2<i1,i2>}:N-C "));
```

Comment:
Was this test case failing? I would have expected that the matcher expression should have taken care of any changes in the canonical form (here: order of summands)?

Reply:
see comments re: hashing. This did expose some leakage of side effects from unit tests and is also fixed in this PR.